src/tool/hpcrun/sample-sources/opencl.c

```diff
@@ -0,0 +1,234 @@
+//
+// --------------------------------------------------------------------------
+// Part of HPCToolkit (hpctoolkit.org)
+//
+// Information about sources of support for research and development of
+// HPCToolkit is at 'hpctoolkit.org' and in 'README.Acknowledgments'.
+// --------------------------------------------------------------------------
+//
+// Copyright ((c)) 2002-2020, Rice University
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are
+// met:
+//
+// * Redistributions of source code must retain the above copyright
+//   notice, this list of conditions and the following disclaimer.
+//
+// * Redistributions in binary form must reproduce the above copyright
+//   notice, this list of conditions and the following disclaimer in the
+//   documentation and/or other materials provided with the distribution.
+//
+// * Neither the name of Rice University (RICE) nor the names of its
+//   contributors may be used to endorse or promote products derived from
+//   this software without specific prior written permission.
+//
+// This software is provided by RICE and contributors "as is" and any
+// express or implied warranties, including, but not limited to, the
+// implied warranties of merchantability and fitness for a particular
+// purpose are disclaimed. In no event shall RICE or contributors be
+// liable for any direct, indirect, incidental, special, exemplary, or
+// consequential damages (including, but not limited to, procurement of
+// substitute goods or services; loss of use, data, or profits; or
+// business interruption) however caused and on any theory of liability,
+// whether in contract, strict liability, or tort (including negligence
+// or otherwise) arising in any way out of the use of this software, even
+// if advised of the possibility of such damage.
+//
+// ******************************************************* EndRiceCopyright *
+
+//******************************************************************************
+// system includes
+//******************************************************************************
+
+#include <alloca.h>
+#include <assert.h>
+#include <ctype.h>
+#include <setjmp.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+#include <ucontext.h>
+#include <stdbool.h>
+
+#include <pthread.h>
+
+#ifndef HPCRUN_STATIC_LINK
+#include <dlfcn.h>
+#endif
+
+
+
+//******************************************************************************
+// libmonitor
+//******************************************************************************
+
+#include <monitor.h>
+
+
+
+//******************************************************************************
+// local includes
+//******************************************************************************
+
+#include "simple_oo.h"
+#include "sample_source_obj.h"
+#include "common.h"
+
+#include <hpcrun/control-knob.h>
```

Comment:
i doubt that all of these header files need to be included.

my strategy for pruning includes is to put an #if 0 ... #endif around all of the includes. try compiling and see what breaks. then add back a header file needed to resolve a reported compiler error. repeat until it compiles. delete the #if 0 ... #endif and all of the includes of headers that weren't necessary to successfully compile.